python/triton/language/standard.py

```diff
@@ -259,11 +259,36 @@ def _sum_combine(a, b):
 # sum
 
 
+def _pick_sum_dtype(in_dtype: core.constexpr, dtype: core.constexpr):
+    # FIXME: Workaround the interpreter not wrapping constexpr args.
+    is_wrapped = isinstance(dtype, core.constexpr)
```

Comment:
fyi @Jokeren when a function is invoked by the JIT compiler, constexpr args are wrapped in a `tl.constexpr`, so this would receive values like `constexpr(None)` or `constexpr(int32)`. The interpreter doesn't wrap/unwrap which necessitates this workaround. Let me know if you have any ideas on how to best address this!

Reply:
If you don't return `constexpr` and use `_unwrap_if_constexpr`, we can simplify the code by removing `is_wrapped`. Might be cleaner.